src/test/java/io/cdap/plugin/splunk/common/util/SearchHelperTest.java

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.plugin.splunk.common.util;
+
+import io.cdap.plugin.splunk.source.batch.SplunkBatchSourceConfig;
+import org.junit.Assert;
+import org.junit.Test;
+
+/**
+ * Tests for {@link SearchHelper}
+ */
+public class SearchHelperTest {
+
+  @Test
+  public void testDecorateSearchStringNull() {
+    SplunkBatchSourceConfig config = new SplunkBatchSourceConfig(
+      "reference",
+      "basic",
+      "apiToken",
+      "userName",
+      60000,
+      60000,
+      3,
+      60000,
+      100,
+      "invalid",
+      "password",
+      "executionMode",
+      "outputFormat",
+      null,
+      "searchId",
+      0L,
+      "earliestTime",
+      "latestTime",
+      "indexedEarliestTime",
+      "indexedLatestTime",
+      100L,
+      "schema");
+
+    String actual = SearchHelper.decorateSearchString(config);
+
+    Assert.assertNull(actual);
+  }
+
+  @Test
+  public void testDecorateSearchStringEmpty() {
+    SplunkBatchSourceConfig config = new SplunkBatchSourceConfig(
+      "reference",
+      "basic",
+      "apiToken",
+      "userName",
+      60000,
+      60000,
+      3,
+      60000,
+      100,
+      "invalid",
+      "password",
+      "executionMode",
+      "outputFormat",
+      "",
+      "searchId",
+      0L,
+      "earliestTime",
+      "latestTime",
+      "indexedEarliestTime",
+      "indexedLatestTime",
+      100L,
+      "schema");
+
+    String actual = SearchHelper.decorateSearchString(config);
+
+    Assert.assertNotNull(actual);
+    Assert.assertTrue(actual.isEmpty());
+  }
+
+  @Test
+  public void testDecorateSearchStringDecorate() {
+    String expected = "search * | kvform";
+
+    SplunkBatchSourceConfig config = new SplunkBatchSourceConfig(
```

Comment:
please use Builder in all tests for config classes as we do in other plugins to make it easy to understand how this invalid plugin is different from valid one

Reply:
Done.